File: 2285-maximum-total-importance-of-roads/2285-maximum-total-importance-of-roads.cpp
class Solution {
public:
    long long maximumImportance(int n, vector<vector<int>>& roads) {
        vector<int> degrees(n,0);
        for(auto &road:roads){
            degrees[road[0]]++;
            degrees[road[1]]++;
        }
        
        sort(degrees.begin(), degrees.end());
        
        long long result=0;
        for(long long i=1; i<=n; i++) result+=degrees[i-1]*i;
        
        return result;
    }
};

File: 2285-maximum-total-importance-of-roads/2285-maximum-total-importance-of-roads.py
class Solution:
    def maximumImportance(self, n: int, roads: List[List[int]]) -> int:
        degrees = [0] * n
        
        for u, v in roads:
            degrees[u] += 1
            degrees[v] += 1
        
        degrees.sort()
        degrees.reverse()
        
        total = 0
        for i, x in enumerate(degrees):
            total += (n - i) * x
        return total